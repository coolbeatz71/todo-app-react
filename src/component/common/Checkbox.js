import PropTypes from 'prop-types';
import React from 'react';

/**
 * Checkbox component
 * @returns {void}
 * @param {object} props
 */
const Checkbox = props => {
  const { checked, name, markComplete, labelClass, spanClass } = props;
  return (
    <label htmlFor="complete" onClick={markComplete} className={labelClass}>
      <input type="checkbox" name={name} checked={checked} />
      <span className={spanClass} />
    </label>
  );
};

Checkbox.propTypes = {
  checked: PropTypes.bool.isRequired,
  name: PropTypes.string,
  markComplete: PropTypes.func.isRequired,
  labelClass: PropTypes.string,
  spanClass: PropTypes.string,
};

Checkbox.defaultProps = {
  name: 'complete',
  labelClass: 'switch',
  spanClass: 'slider round',
};

export default Checkbox;
