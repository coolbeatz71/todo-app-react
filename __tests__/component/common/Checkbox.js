import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../../../src/component/common/Checkbox';

const props = {
  checked: false,
  markComplete: jest.fn(),
};
describe('Checkbox.js', () => {
  test('should render the checkbox component', () => {
    const wrapper = shallow(<Checkbox {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
