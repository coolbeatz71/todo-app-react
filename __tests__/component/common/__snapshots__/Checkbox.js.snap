// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.js should render the checkbox component 1`] = `ShallowWrapper {}`;
